web/app/page.js

```diff
@@ -58,5 +64,40 @@ export default function Home() {
 }
 
 function allergenTimeSortCallback(a, b) {
-  return b.time - a.time;
+  if (a.time && b.time) {
+    return a.time - b.time;
+  }
+  if (!a.time) {
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
+function loadJSONfromLocal(key) {
+  const storedTimes = JSON.parse(window.localStorage.getItem(key));
+  if (storedTimes === null) {
+    console.log("LoadJSON reading null");
+    return null;
+  }
+  return storedTimes.map(({ name, time }) => {
+    if (time) {
+      return { name, time: new Date(time) };
+    }
+    return { name, time };
+  });
+}
+
+function storeJSONinLocal(key, value) {
+  window.localStorage.setItem(key, JSON.stringify(value));
+}
+
+function stateSetter() {
+  const allergens = loadJSONfromLocal("allergens");
+  if(allergens) {
+    return allergens
+  }
+  return allergenNames.map(({name}) => {
+    return {name, time: null}
+  })
```

Comment:
Since `allergens` will either be an object or null, we can get away with this format to default to our original array if it's not retrieved
```suggestion
  return allergens || allergenNames.map(({name}) => {
    return {name, time: null}
  })
```

Reply:
cool!